app/src/test/java/com/github/se/travelpouch/model/activity/map/DirectionsRepositoryTest.kt

```diff
@@ -35,17 +35,34 @@ class DirectionsRepositoryTest {
     // Mocking the response body
     val responseBody =
         """
-            {
-                "status": "OK",
-                "routes": [
-                    {
-                        "overview_polyline": {
-                            "points": "encodedPolylineString"
+        {
+            "status": "OK",
+            "routes": [
+                {
+                    "overview_polyline": {
+                        "points": "encodedPolylineString"
+                    },
+                    "legs": [
+                        {
+                            "distance": { "text": "5 km", "value": 5000 },
+                            "duration": { "text": "10 mins", "value": 600 },
+                            "start_address": "Origin Address",
+                            "end_address": "Destination Address",
+                            "start_location": { "lat": 1.0, "lng": 2.0 },
+                            "end_location": { "lat": 3.0, "lng": 4.0 },
+                            "steps": [
+                                {
+                                    "polyline": {
+                                        "points": "u{~vFvyys@fC_y@"
+                                    }
+                                }
+                            ]
```

Comment:
I suppose that is a json file. You can in a future improvement of this PR put this in another file .json and call this file in the test.

Reply:
It is a good idea. I will think about it